### PR TITLE
Add comp code

### PR DESCRIPTION
**Pull request by chamezopoulos**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,120 @@
 # DAGPap24
+
 This repo includes all content relevant to DAGPap24 (Detection of Artificially Generated Scientific Papers 2024) competition, organised for SDP 2024
```
Comment (Yorko) [inline README.md]: Great readme. I'd only add a link to the competition right in the first sentence :)

File: config/config_baseline.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  train_file_name: data_gen_content_train_toy.json # this is the train file to be used by hugginface_ner.py
+  validation_file_name: data_gen_content_val_toy.json # this is the validation file to be used by hugginface_ner.py
+  test_file_name: data_gen_content_dev_toy.json # this is the test file to be used by hugginface_ner.py
+environment:
+  SEED: 0
+  val_size: 0.2 # this is used in span_prediction to split train dataset 
+bert: # bert model params, used in huggingface_ner.py
+  model: distilbert/distilbert-base-uncased
+  MAX_LENGTH: 512
+  script_file_location : src/ml_gen_detection/huggingface_ner.py
```
Comment (Yorko) [inline config/config_baseline.yml]: Do we need this `src/ml_gen_detection/huggingface_ner.py`? Looks like legacy. If we need it, then should be `hf_token_classification`

File: config/config_baseline.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+data:
+  path_to_data: data
+  train_file: train_data_toy.parquet # this is for the train_test_split.py output
+  test_file: dev_data_toy.parquet # this is for the train_test_split.py and eval_f1.py output
```
Comment (Yorko) [inline config/config_baseline.yml]: I suggest specifying the exact same params reproducing the baseline:

- test_file: dev_data.parquet
- num_train_epochs: 20
- etc.

File: config/config_baseline.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  train_file_name: data_gen_content_train_toy.json # this is the train file to be used by hugginface_ner.py
+  validation_file_name: data_gen_content_val_toy.json # this is the validation file to be used by hugginface_ner.py
+  test_file_name: data_gen_content_dev_toy.json # this is the test file to be used by hugginface_ner.py
+environment:
+  SEED: 0
+  val_size: 0.2 # this is used in span_prediction to split train dataset 
+bert: # bert model params, used in huggingface_ner.py
```
Comment (Yorko) [inline config/config_baseline.yml]: we don't have `huggingface_ner.py` anymore, the comments need to be adjusted as well

File: README.md [hunk 5]
```diff
@@ -1,2 +1,120 @@
...
+
+zip my_new_submission.zip predictions.parquet
+
+Please note that you should not compress the folder with the prediction files. Instead, you need to compress the files themselves: i.e., if you are submitting from Macbook (and you do not want to compress via terminal), you need to select all the prediction files that you want to send and press "Compress".
+
+3. Submit your solution via the My submissions tab.
+
+## Baseline
+
+We're also providing a baseline where we use [DistilBERT](https://huggingface.co/docs/transformers/model_doc/distilbert) to approach the task as an NER task. To run the baseline, execute:
```
Comment (Yorko) [inline README.md]:
```suggestion
We're also providing a baseline where we use [DistilBERT](https://huggingface.co/docs/transformers/model_doc/distilbert) to approach the task as an NER task. To run the baseline, execute:
```

We're also providing a baseline where we use [DistilBERT](https://huggingface.co/docs/transformers/model_doc/distilbert) to approach the task as a token classification task. To run the baseline, execute:

File: README.md [hunk 6]
```diff
@@ -1,2 +1,120 @@
...
+- 0 for 'human'
+- 1 for 'NLTK_synonym_replacement'
+- 2 for 'chatgpt'
+- 3 for 'summarized'
+
+See "[sample_dev_submission.zip](https://drive.google.com/file/d/1Xm59UlLJ-aemDG-PZ3SCBJa_5EkzGLXq)" (from the [Data page](https://www.codabench.org/competitions/2431/#/pages-tab)) for an example. It is an all-zero submission that corresponds to predicting that all text is human-written.
+
+2. Compress (zip) the prediction file, for example, with a bash command:
+
+zip my_new_submission.zip predictions.parquet
```
Comment (Yorko) [inline README.md:104]: could be formatted as code

File: src/ml_gen_detection/dagpap24_baseline.py [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+
+    return pd.DataFrame(
+        {"index": index_list, "tokens": tokens_list, "labels": labels_list}
+    )
+
+
+def write_df_to_json(df: pd.DataFrame, path_to_json: str):
+    """
+    This function writes pandas dataframes into a compatible json format
+    to be used by huggingface_ner
```
Comment (Yorko) [inline src/ml_gen_detection/dagpap24_baseline.py]:
```suggestion
    to be used by huggingface_ner
```
to be used by `hf_token_classification`

**Review by Yorko — CHANGES_REQUESTED** [review]
Great job!

Only minor polishing suggested, see comments.

File: README.md (hunk 1 above)
Comment (chamezopoulos) [reply README.md]: Fixed that

File: config/config_baseline.yml (hunk 2 above)
Comment (chamezopoulos) [reply config/config_baseline.yml]: No, we don't. Removed this.

**Review by Yorko — APPROVED** [review]
Looks good, hit the green button!

**Pull request merged**